cvmfs/catalog_mgr_ro.cc

```diff
@@ -60,6 +102,22 @@ LoadReturn SimpleCatalogManager::LoadCatalogByHash(
                       url.c_str(), retval, download::Code2Ascii(retval));
   }
 
+  // for writable catalog make copy in dir_temp_ that can be modified
+  if (use_local_cache_ && copy_to_tmp_dir_) {
+    std::string tmp_path = ctlg_context->sqlite_path();
+    std::string cache_path = tmp_path;
+    fcatalog = CreateTempFile(dir_temp_ + "/catalog", 0666, "w", &tmp_path);
+    if (!fcatalog) {
+      PANIC(kLogStderr, "failed to create temp file when loading %s",
+                        url.c_str());
+    }
+    ctlg_context->SetSqlitePath(tmp_path);
+
+    CopyPath2File(cache_path, fcatalog);
+    fclose(fcatalog);
+  }
+
+  ctlg_context->SetHash(effective_hash);
```

Comment:
i think we can remove this because `effective_hash` is only set in the beginning and no download/whatever action changes it?

Reply:
this was the somewhat tricky part of the function before the refactor to use `catalog_context` which replaces `catalog_path` and `catalog_hash`

```c++
 FILE *fcatalog;

  if (use_local_cache_) {
    std::string tmp_path = local_cache_dir_ + "/"
                   + effective_hash.MakePathWithoutSuffix();

    *catalog_path = local_cache_dir_ + "/"
                   + effective_hash.MakePathWithoutSuffix();

    // file is cached
    if (FileExists(catalog_path->c_str())) {
      if (!copy_to_tmp_dir_) {
        *catalog_hash = effective_hash;
        return kLoadNew;
      } else {  // for writable catalog create copy in dir_temp_
        std::string cache_path = *catalog_path;
        fcatalog = CreateTempFile(dir_temp_ + "/catalog", 0666, "w",
                                                                  catalog_path);
        if (!fcatalog) {
          PANIC(kLogStderr, "failed to create temp file when loading %s",
                url.c_str());
        }

        CopyPath2File(cache_path, fcatalog);
        fclose(fcatalog);

        *catalog_hash = effective_hash;
        return kLoadNew;
      }
    }

    fcatalog = fopen(catalog_path->c_str(), "w");
  } else {  // use tmp file
    fcatalog = CreateTempFile(dir_temp_ + "/catalog", 0666, "w", catalog_path);
    if (!fcatalog) {
      PANIC(kLogStderr, "failed to create temp file when loading %s",
            url.c_str());
    }
  }
```